File: MotivApp/app/src/main/java/com/motiv/piotr/Post.java
package com.motiv.piotr;

import androidx.annotation.NonNull;
import com.google.gson.*;
import com.google.gson.annotations.*;
import com.google.gson.reflect.*;
import io.realm.*;
import java.util.*;
import java.util.concurrent.*;

public class Post extends RealmObject {

    private static final Gson gson = new Gson();
    @NonNull private java.lang.String id = UUID.randomUUID().toString();

    @SerializedName("user_id")
    private java.lang.String user_id;

    @SerializedName("_links")
    private com.motiv.piotr.Links links;

    @SerializedName("title")
    private java.lang.String title;

    @SerializedName("body")
    private java.lang.String body;

    public java.lang.String getId() {

        return this.id;
    }

    public void setId(java.lang.String id) {
        this.id = id;
    }

    public java.lang.String getUser_id() {

        return this.user_id;
    }

    public void setUser_id(java.lang.String user_id) {
        this.user_id = user_id;
    }

    public com.motiv.piotr.Links getLinks() {

        return this.links;
    }

    public void setLinks(com.motiv.piotr.Links links) {
        this.links = links;
    }

    public java.lang.String getTitle() {

        return this.title;
    }

    public void setTitle(java.lang.String title) {
        this.title = title;
    }

    public java.lang.String getBody() {

        return this.body;
    }

    public void setBody(java.lang.String body) {
        this.body = body;
    }

    public static Post fromJson(String json) {
        return gson.fromJson(json, Post.class);
    }

    public static String toJson(Post object) {
        return gson.toJson(object, Post.class);
    }

    public static String toJsonArray(Post[] array) {
        return gson.toJson(array, Post[].class);
    }

    public static Post[] fromJsonArray(String array) {
        return gson.fromJson(array, Post[].class);
    }
}

File: MotivApp/app/src/main/java/com/motiv/piotr/Post.kt
 
package com.motiv.piotr
import androidx.annotation.NonNull
import com.google.gson.*
import com.google.gson.annotations.*
import com.google.gson.reflect.*
import io.realm.*
import java.util.*
import java.util.concurrent.*

public open class Post : RealmObject() {

    @NonNull private var id: String = UUID.randomUUID().toString()

    @SerializedName("user_id")

    private var user_id: String = ""

    @SerializedName("_links")

    private var links: com.motiv.piotr.Links = com.motiv.piotr.Links()

    @SerializedName("title")

    private var title: String = ""

    @SerializedName("body")

    private var body: String = ""

    fun getId(): String {
        return this.id
    } fun setId(id: String) {
        this.id = id
    } fun getUser_id(): String {
        return this.user_id
    } fun setUser_id(user_id: String) {
        this.user_id = user_id
    } fun getLinks(): com.motiv.piotr.Links {
        return this.links
    } fun setLinks(links: com.motiv.piotr.Links) {
        this.links = links
    } fun getTitle(): String {
        return this.title
    } fun setTitle(title: String) {
        this.title = title
    } fun getBody(): String {
        return this.body
    } fun setBody(body: String) {
        this.body = body
    } companion object {
        val gson: Gson = Gson()
        fun fromJson(json: String): Post {
            return gson.fromJson(json, Post::class.java)
        }

        fun toJson(obj: Post): String {
            return gson.toJson(obj, Post::class.java)
        }

        fun toJsonArray(obj: Post): String {
            return gson.toJson(obj, Array<Post>::class.java)
        }

        fun fromJsonArray(json: String): Array<Post> {
            return gson.fromJson(json, Array<Post>::class.java)
        }
    }
}
